lib/AST/Type.cpp

```diff
@@ -4168,15 +4168,20 @@ ClangTypeInfo AnyFunctionType::getCanonicalClangTypeInfo() const {
 ASTExtInfo
 AnyFunctionType::getCanonicalExtInfo(bool useClangFunctionType) const {
   assert(hasExtInfo());
+
+  CanGenericSignature genericSig;
+  if (auto *genericFnTy = dyn_cast<GenericFunctionType>(this))
+    genericSig = genericFnTy->getGenericSignature().getCanonicalSignature();
```

Comment:
getCanonicalSignature() doesn't change the reduced type computation :)

Reply:
I totally copy-pasta'd this. We can remove it from both here and where I got it from (canonicalization of function types)